core/blockchain_impl.go

```diff
@@ -899,7 +904,8 @@ func (bc *BlockChainImpl) Stop() {
 	bc.scope.Close()
 	close(bc.quit)
 	atomic.StoreInt32(&bc.procInterrupt, 1)
-
+	// wait pruning completed
+	bc.waitPruning(false)
```

Comment:
Should this be done after the `HEAD`, `HEAD-1` and `HEAD-127` ?

Reply:
no, here is waiting for the previous pruning to complete, we must ensure the pruning is finished before next state writing.